test/unit/spec/MfaVerify_spec.js

```diff
@@ -170,15 +170,36 @@ function (Q,
       ]);
     }
 
+    function mockChrome(isAvailable){
+      BrowserFeatures.isChrome = function () {};
+      spyOn(BrowserFeatures, 'isChrome').and.returnValue(isAvailable);
+    }
+
+    function mockFirefox(isAvailable){
+      BrowserFeatures.isFirefox = function () {};
+      spyOn(BrowserFeatures, 'isFirefox').and.returnValue(isAvailable);
+    }
 
     function setupU2F(options) {
-      if (!window.u2f || !window.u2f.sign) {
-        window.u2f = { sign: function () {} };
+      options || (options = {});
+
+      mockChrome(!options.noChrome);
```

Comment:
I don't understand the duplication problem you are talking about... but I still think that it is easier to understand a simple boolean than a double negation
```if (options.u2f) {```vs ```if (!options.noU2f) {```
or 
```setupU2F({u2f: false, firefox: false})``` vs ```setupU2F({noU2f: true, noFirefox: true})```